completion/all.go

```diff
@@ -6,74 +6,81 @@ import (
 	"os"
 	"os/user"
 	"path/filepath"
+	"runtime"
 	"strings"
 	"text/template"
 
 	"github.com/coder/serpent"
 )
 
+type Shell interface {
+	Name() string
```

Comment:
There's no other way to get the name of the detected shell, otherwise.

Reply:
this is a common idiom, dw